pkg/network/infraconfigurators/common.go

```diff
@@ -48,3 +49,14 @@ func isMultiqueue(vmi *v1.VirtualMachineInstance) bool {
 	return (vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue != nil) &&
 		(*vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue)
 }
+
+func retrieveMacAddressFromVMISpecIface(vmiSpecIface *v1.Interface) (*net.HardwareAddr, error) {
+	if vmiSpecIface.MacAddress != "" {
+		macAddress, err := net.ParseMAC(vmiSpecIface.MacAddress)
+		if err != nil {
+			return nil, err
+		}
+		return &macAddress, nil
+	}
+	return nil, nil
+}
```

Comment:
I think this function does not belong in this file - imo, it is meant to hold everything netlink related.

Reply:
Just looking on the methods beyond the added one we can see this rule of "hold everything netlink related" is not applied (`isMultiqueue`, `calculateNetworkQueues`, `validateMTU`, `generateTapDeviceName`).